charts/spire/charts/spire-agent/templates/configmap.yaml

```diff
@@ -116,6 +116,9 @@ plugins:
         disable_container_selectors: {{ .Values.workloadAttestors.k8s.disableContainerSelectors }}
         use_new_container_locator: {{ .Values.workloadAttestors.k8s.useNewContainerLocator }}
         verbose_container_locator_logs: {{ .Values.workloadAttestors.k8s.verboseContainerLocatorLogs }}
+        {{- if eq (include "spire-agent.connect-by-hostname" .) "true" }}
+        node_name_env: "MY_NODE_NAME"
+        {{- end }}
```

Comment:
According to https://github.com/spiffe/spire/blob/main/doc/plugin_agent_workloadattestor_k8s.md
node_name_env is defaulted to MY_NODE_NAME when unset.

So, I think this is a  NOOP.

But having it explicit is always nice.

I'm ok with merging this.

Reply:
I tested this change on Openshift cluster and it works fine.